### PR TITLE
fix: escape spaces in onerepo cli installation path

### PR DESCRIPTION
**Pull request by jakeleveroni**
**Problem:**
When installing the onerepo cli, if the user installs while under a path that includes spaces the installation fails to escape the spaces breaking the command 

**Solution:**
Escape any spaces in the installation path during install

### CODE REVIEW

File: modules/test-cli/src/index.ts [hunk 1]
```diff
@@ -21,14 +21,25 @@ const testRunner: typeof vitest | typeof jest =
...
 
 	process.env.ONE_REPO_VERBOSITY = '4';
 	process.env.ONE_REPO_HEAD_BRANCH = 'main';
-	process.argv[1] = 'onerepo-test-runner';
+	process.env = {
+		...process.env,
+		ONE_REPO_VERBOSITY: '4',
+		ONE_REPO_HEAD_BRANCH: 'main',
+		...(builderExtras?.env ?? {}),
+	};
```
Comment (paularmstrong) [inline modules/test-cli/src/index.ts:40]: De-duplicate the first two defaults from line 33,34

File: modules/core/src/core/install/commands/install.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export const handler: Handler<Args> = async function handler(argv, { graph, logg
 	await sudo({
 		name: 'Create executable',
 		cmd: 'echo',
-		args: [`"#!/bin/zsh\n\n${process.argv[1]} \\$@"`, '|', 'sudo', 'tee', installLocation],
+		args: [`"#!/bin/zsh\n\n${process.argv[1].replace(' ', '\\ ')} \\$@"`, '|', 'sudo', 'tee', installLocation],
```
Comment (FXStrato) [inline modules/core/src/core/install/commands/install.ts:103]: doesn't this replace only the first space? would need to add `/g` i believe to make it replace all open spaces

File: modules/test-cli/src/index.ts [hunk 3]
```diff
@@ -129,17 +140,27 @@ export async function runHandler<R = Record<string, unknown>>(
...
-type Extras = Omit<HandlerExtra, 'getAffected' | 'getFilepaths' | 'getWorkspaces' | 'logger'>;
+type Extras = { builderExtras?: BuilderExtras } & Omit<
+	HandlerExtra,
+	'getAffected' | 'getFilepaths' | 'getWorkspaces' | 'logger'
+>;
+
+type BuilderExtras = {
+	argv?: Array<string>;
+	env?: Exclude<typeof process.env, 'clear'>;
+};
```
[the remark this answers is not in the document]
Comment (jakeleveroni) [reply modules/test-cli/src/index.ts]: yeah this makes it way cleaner, good call!

**Review by paularmstrong — APPROVED** [review]
Nice, thanks!!

**Pull request merged**